web-client/public/js/update-app.js

```diff
@@ -48,6 +51,28 @@ export const updateApp = grnState => {
         grnState.newNetwork = false;
     }
 
+
+    if (grnState.normalizationMax !== null) {
+        var valueValidator = (min, max, value) => {
+            return Math.min(max, Math.max(min, value));
+        };
+
+        var edgeWeightNormalizationInputValidation = (value) => {
+            return value ===
+            "" ? "" : valueValidator(MIN_EDGE_WEIGHT_NORMALIZATION, MAX_EDGE_WEIGHT_NORMALIZATION, value);
+        };
+
+        var synchronizeNormalizationValues = (value) => {
+            var validated = edgeWeightNormalizationInputValidation(value);
+            $("#normalization-max").val(validated);
+            $("#edge-weight-normalization-factor-menu").val(validated);
+        };
```

Comment:
Is there a good reason for embedding `valueValidator`, `edgeWeightNormalizationInputValidation`, and `synchronizeNormalizationValues` inside the `if` statement? I don't see any dependencies on the function's scope. At this location, these functions get defined over and over again whenever `updateApp` is called. If there is a scope dependency, then that is tolerable, but I don't see that here. Move these outside, the way `displayNetwork` is outside. That way they only get defined once.